packages/cozy-harvest-lib/src/models/ConnectionFlow.js

```diff
@@ -624,12 +637,32 @@ export class ConnectionFlow {
     }
   }
 
+  watchTriggerJobs() {
+    // can happen when the current trigger comes from realtime
+    if (this.trigger) {
+      if (!this.trigger?.current_state) {
+        this.refetchTrigger()
+      }
+      this.realtime.subscribe(
+        'updated',
+        JOBS_DOCTYPE,
+        this.handleTriggerJobUpdated
+      )
+    }
+  }
+
   watchJob(options = { autoSuccessTimer: false }) {
+    if (this.job?._id === this.jobWatcher?.job?._id) {
+      return
+    }
+    if (this.jobWatcher) {
+      this.jobWatcher.unsubscribeAll()
+    }
```

Comment:
Can you explain those both changes please?

Reply:
What is the purpose of `watchJob`, `watchKonnectorJob` and `watchTriggerJobs`? 

Why don't we subscribe directly to realtime event inside the constructor?

Reply:
I added jsdoc to watchJob now but here it is :

* watchJob uses watchKonnectorJob to watch a specific job, it also relays events from KonnectorJobWatcher instance to the current ConnectionFlow instance (like LOGIN_SUCCESS after 8 seconds)
* watchKonnectorJob creates a new instance of KonnectorJobWatcher to watch a specified job with a given id. It also handle the autoSucessFullTimer option which triggers LOGIN_SUCCESS after 8 second when true
* watchTriggerJobs : detects new jobs associated to the current trigger. If any, it refetches the current trigger which will set the new job as the current job to watch with watchJob